packages/datadog-plugin-jest/src/index.js

```diff
@@ -58,6 +64,88 @@ class JestPlugin extends Plugin {
     this.testEnvironmentMetadata = getTestEnvironmentMetadata('jest', this.config)
     this.codeOwnersEntries = getCodeOwnersFileEntries()
 
+    this.addSub('ci:jest:session:start', (command) => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+      const store = storage.getStore()
+      const childOf = getTestParentSpan(this.tracer)
+      const testSessionSpanMetadata = getTestSessionCommonTags(command, this.tracer._version)
+
+      const testSessionSpan = this.tracer.startSpan('jest.test_session', {
+        childOf,
+        tags: {
+          ...this.testEnvironmentMetadata,
+          ...testSessionSpanMetadata
+        }
+      })
+      this.enter(testSessionSpan, store)
+    })
+
+    this.addSub('ci:jest:session:finish', (status) => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+      const testSessionSpan = storage.getStore().span
+      testSessionSpan.setTag(TEST_STATUS, status)
+      testSessionSpan.finish()
+      finishAllTraceSpans(testSessionSpan)
+      this.tracer._exporter._writer.flush()
+    })
+
+    // Test suites can be run in a different process from jest's main one.
+    // This subscriber changes the configuration objects from jest to inject the trace id
+    // of the test session to the processes that run the test suites.
+    this.addSub('ci:jest:session:configuration', configs => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+      const testSessionSpan = storage.getStore().span
+      configs.forEach(config => {
+        config._ddTestSessionId = testSessionSpan.context()._traceId.toString(16)
+        config._ddTestCommand = testSessionSpan.context()._tags[TEST_COMMAND]
+      })
+    })
+
+    this.addSub('ci:jest:test-suite:start', ({ testSuite, testEnvironmentOptions }) => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+
+      const { _ddTestSessionId: testSessionId, _ddTestCommand: testCommand } = testEnvironmentOptions
+
+      const store = storage.getStore()
+
+      const testSessionSpanContext = this.tracer.extract('text_map', {
+        'x-datadog-trace-id': id(testSessionId).toString(10),
+        'x-datadog-span-id': id(testSessionId).toString(10),
```

Comment:
What is `x-datadog-span-id`?

Reply:
yeah this slipped. Removing it

Reply:
done in https://github.com/DataDog/dd-trace-js/pull/2122/commits/24a2f1bcce959a6158ae71c9a49475c80b55a147